### PR TITLE
Add CgroupDriver to docker info

### PR DESCRIPTION
**Pull request by hqhq**
Fixes: #19539

Signed-off-by: Qiang Huang h.huangqiang@huawei.com

### CODE REVIEW

**Comment by hqhq** [comment]
Will rebase after https://github.com/docker/engine-api/pull/95 merged.

**Comment by thaJeztah** [comment]
Thanks @hqhq don't forget to update the example output in the documentation;
https://github.com/docker/docker/blob/master/docs/reference/commandline/info.md
and the man-page;
https://github.com/docker/docker/blob/master/man/docker-info.1.md


The API documentation and changes;
https://github.com/docker/docker/blob/master/docs/reference/api/docker_remote_api.md#v123-api-changes
https://github.com/docker/docker/blob/master/docs/reference/api/docker_remote_api_v1.23.md#display-system-wide-information

**Comment by calavera** [comment]
It doesn't make sense to print this in Windows because it's very Linux specific.

**Comment by hqhq** [comment]
@thaJeztah Thanks, will do.

@calavera Didn't test on windows, but it returns empty string on windows, and I think `ioutils.FprintfIfNotEmpty` won't print anything with empty string input, no?

**Comment by hqhq** [comment]
Updated.

File: docs/reference/api/docker_remote_api.md [hunk 1]
```diff
@@ -117,6 +117,7 @@ This section lists each version from latest to oldest.  Each listing includes a
 
 * `GET /containers/json` returns the state of the container, one of `created`, `restarting`, `running`, `paused`, `exited` or `dead`.
 * `GET /networks/(name)` now returns an `Internal` field showing whether the network is internal or not.
+* `GET /info` now returns `CgroupDriver` field showing what cgroup driver is daemon using.
```
Comment (thaJeztah) [inline docs/reference/api/docker_remote_api.md]: Perhaps mention the possible options (like we do above for "state")

```Markdown

 `GET /info` now returns `CgroupDriver` field showing what cgroup driver the daemon is using; `cgroupfs` or `systemd`
```

**Comment by thaJeztah** [comment]
Design looks good to me; moving to code review

File: docs/reference/api/docker_remote_api.md (hunk 1 above)
Comment (hqhq) [inline docs/reference/api/docker_remote_api.md]: Done, thanks.

**Comment by tiborvass** [comment]
@hqhq needs a rebase

**Comment by hqhq** [comment]
@tiborvass Rebased, thanks.

**Comment by thaJeztah** [comment]
@hqhq looks like this still needs a vendor update

**Comment by icecrime** [comment]
@hqhq Please look into CI failures!

```
01:48:50 ---> Making bundle: validate-vendor (in bundles/1.11.0-dev/validate-vendor)
01:51:38 The result of ./hack/vendor.sh differs
01:51:38 
01:51:38  M vendor/src/github.com/docker/engine-api/types/types.go
```

**Comment by hqhq** [comment]
@icecrime Thanks, I've noticed that, have been waiting the update of engine-api for a while and it's still not updated.

**Comment by hqhq** [comment]
I'm doing the update: https://github.com/docker/docker/pull/20774

**Comment by hqhq** [comment]
@thaJeztah @tiborvass Rebased, PTAL.

**Comment by calavera** [comment]
LGTM

**Comment by thaJeztah** [comment]
LGTM (code and docs)

**Comment by thaJeztah** [comment]
ping @MHBauer @vdemeester PTAL

**Comment by vdemeester** [comment]
LGTM 🐯

**Pull request merged**